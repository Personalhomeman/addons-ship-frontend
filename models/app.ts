export type App = {
  appSlug: string;
  plan: string;
  title: string;
  avatarUrl: string;
  projectType: ProjectType;
  androidErrors: string[];
  iosErrors: string[];
  colors?: {
    start?: string;
    end?: string;
  };
};

export type ProjectType =
  | 'xamarin'
  | 'ios'
  | 'osx'
  | 'macos'
  | 'android'
  | 'cordova'
  | 'ionic'
  | 'react-native'
  | 'flutter'
  | 'other';
