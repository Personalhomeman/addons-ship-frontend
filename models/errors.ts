export class RequestError extends Error {
  constructor(public status: number, public statusText: string) {
    super(`${status}: ${statusText}`);
  }
}
