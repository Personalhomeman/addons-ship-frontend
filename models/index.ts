import { NextContext } from 'next';
import { Store } from 'redux';
import { Request } from 'express';

export type AppVersionProps = {
  appSlug: string;
  versionId: string;
};

export const AppVersionPageTabs = ['details', 'devices', 'test', 'activity'];

export type AppVersionPageQuery = AppVersionProps & {
  isPublic?: string;
  selectedTab?: typeof AppVersionPageTabs[number];
};

export type AppPageQuery = {
  appSlug: string;
};

export type AppVersionList = Array<{
  version: string;
  appVersions: AppVersion[];
}>;

export type AppVersion = {
  id: number;
  version: string;
  platform: string;
  buildNumber: number;
  buildSlug: string;
  lastUpdate: Date;
  description: string;
  whatsNew: string;
  minimumOs: string;
  minimumSdk: string;
  packageName: string;
  certificateExpires: Date;
  bundleId: string;
  size: number;
  supportedDeviceTypes: string[];
  distributionType: string;
  iconUrl: string;
  appName: string;
  promotionalText: string;
  keywords: string;
  reviewNotes: string;
  supportUrl: string;
  marketingUrl: string;
  scheme: string;
  configuration: string;
  publicInstallPageURL: string;
};

export interface PageContext extends NextContext {
  store: Store;
  isServer: boolean;
  req: Request;
}

export type Platform = 'ios' | 'android';
