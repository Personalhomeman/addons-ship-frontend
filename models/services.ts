export type APIConfig = {
  url: string;
};

export type AppConfig = {
  baseUrl: string;
};
