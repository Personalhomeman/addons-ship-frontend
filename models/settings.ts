export const AppSettingsPageTabs = ['general', 'notifications'];

export type AppSettingsPageQuery = {
  appSlug: string;
  selectedTab?: typeof AppSettingsPageTabs[number];
};

export type IosSettings = {
  artifactExposingWorkflows: string;
  appleDeveloperAccountEmail: string;
  appSku: string;
  appSpecificPassword: string;
  selectedProvProfile: any;
  selectedCertificate: any;
};

export type AndroidSettings = {
  artifactExposingWorkflows: string;
  track: string;
  selectedKeystoreFile: any;
  selectedServiceAccountJsonFile: any;
};

export type ProvProfile = { name: string };
export type Certificate = { name: string };
export type KeystoreFile = { name: string };
export type ServiceAccountJsonFile = { name: string };

export type Settings = {
  iosSettings?: IosSettings;
  androidSettings?: AndroidSettings;
  provProfiles?: ProvProfile[];
  certificates?: Certificate[];
  keystoreFiles?: KeystoreFile[];
  serviceAccountJsonFiles?: ServiceAccountJsonFile[];
  projectType:
    | 'xamarin'
    | 'ios'
    | 'osx'
    | 'macos'
    | 'android'
    | 'cordova'
    | 'ionic'
    | 'react-native'
    | 'flutter'
    | 'other';
};

export type AppContact = {
  id: string;
  email: string;
  isConfirmed: boolean;
  notificationPreferences: {
    newVersion: boolean;
    successfulPublish: boolean;
    failedPublish: boolean;
  };
  isMarkedForDelete?: boolean;
  isMarkedForUpdate?: boolean;
};
