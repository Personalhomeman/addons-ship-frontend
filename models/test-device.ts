export type TestDevice = {
  deviceId: string;
  deviceType: string;
  owner: string;
};
