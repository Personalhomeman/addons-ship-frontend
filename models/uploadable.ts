export type Uploadable = {
  filename: string;
  filesize: number;
  deviceType: string;
  screenSize: string;
  uploadUrl?: string;
};
