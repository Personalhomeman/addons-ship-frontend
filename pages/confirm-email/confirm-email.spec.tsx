jest.mock('@/services/ship-api');

import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { PageContext } from '@/models';
import shipApi from '@/services/ship-api';

import ConfirmEmail, { Props } from '.';

describe('Confirm Email', () => {
  const defaultProps: Props = {
    app: {
      appSlug: 'app-slug',
      title: 'The Best App Ever',
      plan: 'gold',
      avatarUrl: 'avatarl.url',
      projectType: 'ios'
    },
    appContact: {
      id: 'bit-bot',
      email: 'bit.bot@bitrise.io',
      isConfirmed: true,
      notificationPreferences: {
        newVersion: true,
        successfulPublish: true,
        failedPublish: true
      }
    }
  };

  describe('it renders without errors', () => {
    test('when it has everything loaded', () => {
      const tree = shallowToJson(shallow(<ConfirmEmail {...defaultProps} />));

      expect(tree).toMatchSnapshot();
    });

    test('when there was an error', () => {
      const tree = shallowToJson(shallow(<ConfirmEmail error="Ouchie!" />));

      expect(tree).toMatchSnapshot();
    });
  });

  describe('getInitialProps', () => {
    it('calls the API with the token', async () => {
      (shipApi.confirmEmail as jest.Mock).mockResolvedValueOnce(defaultProps);
      const token = 'this-is-a-confirmation-token';

      const props = await ConfirmEmail.getInitialProps(({ query: { token } } as any) as PageContext);

      expect(shipApi.confirmEmail).toHaveBeenCalledWith(token);
      expect(props).toEqual(defaultProps);
    });

    it('returns an error if the API had an issue', async () => {
      (shipApi.confirmEmail as jest.Mock).mockRejectedValueOnce('api had some issue');

      const props = await ConfirmEmail.getInitialProps(({ query: { token: 'whatever' } } as any) as PageContext);
      expect(props).toEqual({
        error: 'Failed to confirm email address'
      });
    });
  });

  test('componentDidMount reloads the page if it loaded incorrectly', () => {
    const { location } = (global as any).window as Window;
    location.reload = jest.fn();

    shallow(<ConfirmEmail />);

    expect(location.reload).toHaveBeenCalled();
  });
});
