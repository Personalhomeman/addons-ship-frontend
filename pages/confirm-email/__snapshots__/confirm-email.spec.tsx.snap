// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Email it renders without exceptions when it has everything loaded 1`] = `
<_default
  appIconUrl="avatarl.url"
  appSlug="app-slug"
  appTitle="The Best App Ever"
  avatarUrl="avatarl.url"
  email="bit.bot@bitrise.io"
  id="bit-bot"
  isConfirmed={true}
  notificationPreferences={
    Object {
      "failedPublish": true,
      "newVersion": true,
      "successfulPublish": true,
    }
  }
  plan="gold"
  projectType="ios"
  title="The Best App Ever"
/>
`;

exports[`Confirm Email it renders without exceptions when there is an error message to show 1`] = `
<Flex
  alignChildrenHorizontal="middle"
  padding="x16"
>
  <Notification
    type="alert"
  >
    Ouchie!
  </Notification>
</Flex>
`;
